### ISSUE TITLE
Failed to activate the rulerz package

### ISSUE THREAD

**Issue by danielbayley** [forgecrafted/rulerz#25]
**Atom**: 1.19.0-beta4 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.5
**Thrown From**: [rulerz](https://github.com/forgecrafted/rulerz) package 0.4.3

### Stack Trace

Failed to activate the rulerz package

```
At Cannot read property 'NaN' of undefined

TypeError: Cannot read property 'NaN' of undefined
    at TextEditorComponent.renderedScreenLineForRow (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:905:31)
    at TextEditorComponent.pixelLeftForRowAndColumn (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:2229:29)
    at TextEditorComponent.pixelPositionForScreenPosition (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:187:40)
    at HTMLElement.pixelPositionForScreenPosition (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-element.js:243:32)
    at HTMLElement.RulerView.update (/packages/rulerz/lib/ruler-view.coffee:61:23)
    at HTMLElement.RulerView.initialize (/packages/rulerz/lib/ruler-view.coffee:32:19)
    at Object.createView (/packages/rulerz/lib/ruler-manager.coffee:25:32)
    at ViewRegistry.module.exports.ViewRegistry.createView (/Applications/Atom Beta.app/Contents/Resources/app/src/view-registry.js:108:78)
    at ViewRegistry.module.exports.ViewRegistry.getView (/Applications/Atom Beta.app/Contents/Resources/app/src/view-registry.js:75:21)
    at CursorModel.module.exports.CursorModel.initialize (/packages/rulerz/lib/cursor-model.coffee:25:25)
    at /packages/rulerz/lib/cursor-model.coffee:20:12)
    at /packages/rulerz/lib/ruler-manager.coffee:33:20
    at TextEditor.module.exports.TextEditor.observeCursors (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor.js:742:9)
    at /packages/rulerz/lib/ruler-manager.coffee:32:49
    at Workspace.observeTextEditors (/Applications/Atom Beta.app/Contents/Resources/app/src/workspace.js:760:9)
    at RulerManager.module.exports.RulerManager.handleEvents (/packages/rulerz/lib/ruler-manager.coffee:30:52)
    at /packages/rulerz/lib/ruler-manager.coffee:20:12)
    at Object.activate (/packages/rulerz/lib/main.coffee:10:35)
    at Package.module.exports.Package.activateNow (/Applications/Atom Beta.app/Contents/Resources/app/src/package.js:251:19)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/package.js:223:32
    at Package.module.exports.Package.measure (/Applications/Atom Beta.app/Contents/Resources/app/src/package.js:97:15)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/package.js:216:26
    at Package.module.exports.Package.activate (/Applications/Atom Beta.app/Contents/Resources/app/src/package.js:213:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:643:34)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:624:29
    at Config.module.exports.Config.transactAsync (/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:344:18)
    at PackageManager.module.exports.PackageManager.activatePackages (/Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:619:19)
    at PackageManager.module.exports.PackageManager.activate (/Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:601:46)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/atom-environment.js:837:30

```

### Commands

```
     -0:14.4.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-when-scrolling)
     -0:06.1.0 env-from-shell:copy (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
```

### Non-Core Packages

```
advanced-open-file 0.16.5 (dev)
aligner 0.22.3 
aligner-css 1.2.0 
aligner-javascript 1.0.0 
aligner-scss 1.1.0 
aligner-stylus 1.0.0 
ansible-snippets 0.2.0 
api-docs 0.0.6 
apm-publish 0.3.0 
asciidoc-preview 2.7.1 
Atom-Bash-Snippets 0.0.3 
atom-beautify 0.28.15 (dev)
atom-calca 0.3.1 
atom-css-class-checker 0.4.3 
atom-css-unit-converter 1.1.1 
atom-github-notifications 0.1.7 
atom-html-preview 0.1.22 
atom-ignore 0.2.0 
atom-jasmine 0.8.1 
atom-jest-snippets 1.0.2 
atom-jq 0.7.0 
atom-live-server 2.1.2 
atom-lupa 0.7.32 
open-terminal-here 2.3.0 
atom-open-this 0.3.0 
atom-react-es2015-snippets 1.1.2 
atom-react-native-autocomplete 0.0.27 
atom-react-native-css 1.1.3 
atom-reverser 2.6.0 
atom-smart-tabs 0.1.0 
atom-sticky-cursor 0.0.8 
atom-terminal-panel 4.4.4 
atom-ternjs 0.18.3 
atom-touch-events 0.23.0 
atom-typescript undefined 
atom-webbrowser 2.1.4 (dev)
atom-xcode 0.1.3 
atom-yeoman 0.3.15 
atomic-spinner 0.0.0 (dev)
auto-detect-indentation 1.3.0 
auto-run 0.2.1 
auto-update-plus 0.2.0 
autoclose-html 0.23.0 
autocomplete-date 0.4.0 
autocomplete-html-entities 0.1.0 
autocomplete-json undefined 
autocomplete-modules 1.6.7 
autocomplete-paths 1.0.5 
autocomplete-ruby 0.2.5 
autocomplete-sass 0.1.0 
autocomplete-swift 1.0.3 
autohide-tree-view 0.25.1 
autoprefixer 3.6.1 
badges 2.2.5 
base-ui undefined (dev)
bezier-curve-editor 0.7.2 
block-and-line-comment 0.4.0 (dev)
browser-plus 0.0.87 (dev)
build 0.65.0 (dev)
build-coffee 0.8.1 
build-decaffeinate 0.1.9 
build-dockerfile 1.0.1 
build-lessc 0.7.1 
build-make 0.13.0 
build-markdown 1.1.0 
build-npm-apm 0.12.0 
build-osa 0.7.2 
build-rake 0.1.1 
build-sassc 0.8.1 
build-stylus 0.8.1 
build-sugarss 0.4.9 
build-xcodebuild 0.2.0 
busy 0.7.0 
busy-signal 1.4.3 
calc 0.4.0 
captain-hook 0.0.1 
change-case 0.6.5 
chmod 0.1.1 
circle-ci 0.10.1 
clean-context-menu 0.3.2 
clipboard-plus 0.5.1 
color-picker 2.1.1 (dev)
context-git 0.14.1 
context-menu-remove 0.1.0 
contextual-preview 0.0.0 (dev)
copy-path-plus 0.1.0 (dev)
css-snippets 1.1.0 
css-spy 0.6.0 
custom-app-icon 0.2.3 (dev)
custom-title 1.0.1 
dash 1.7.0 
default-language 0.3.0 
dict 0.2.3 
dictionary 0.5.0 
disable-grammars 0.1.1 
disable-keybindings 0.6.0 
disable-menus 0.0.0 (dev)
distribute-panes 0.1.0 
dockerletion 0.2.0 
double-tag 0.4.0 
dracula-theme 2.0.0 
duotone-dark-sea-syntax 2.1.0 
duotone-dark-syntax 2.1.0 
duplicate-line-or-selection 0.9.0 
eclectic-syntax 0.2.1 
editorconfig 2.2.0 
emmet 2.4.3 
env-from-shell 0.3.0 
es6-javascript 1.0.0 
esformatter 2.2.1 
expand-region 0.2.6 
expose 0.14.0 
external-open 0.6.3 
fall-syntax 0.0.0 (dev)
file-icons 2.0.15 
file-templates 0.2.0 
file-type-icons 1.3.3 
file-types 0.5.5 
finder-fixes undefined (dev)
firewatch-syntax 1.1.1 
flex-tool-bar 0.12.0 
flexbox-snippets 0.3.0 
flexible-ui 0.10.7 
flow-ide 1.4.2 
fonts 3.0.2 
gfm-pdf 0.3.4 
ghost 0.0.0 (dev)
gist 0.3.1 
git-diff-details 1.4.0 
git-history 3.2.0 
git-plus 7.4.0 (dev)
git-split-diff-hyperclick 0.3.11 
git-travel 0.1.4 
github-clone-desktop-button undefined (dev)
gitkraken 0.4.3 (dev)
goto-definition 1.3.3 
gruvbox 1.0.7 
hey-pane 0.2.0 
highlight-selected 0.12.0 
html-id-class-snippets 1.4.1 
html-to-css 0.2.2 
html2jade-plus 1.0.0 
html2slim 0.4.1 
hyperclick 0.1.2 
hyperclick-octolinker 0.1.0 
hyperlink-hyperclick 1.3.4 
icon-fonts 2.13.0 
imdone-atom 2.1.19 (dev)
imdone-atom-github 0.1.11 
indent-guide-improved 1.4.13 
intentions 1.1.2 
isotope-light-ui 2.2.6 
isotope-ui 2.7.1 
jQuery 1.2.5 
js-hyperclick 1.10.1 
js-patterns-atom-snippets 0.2.1 
language-ansible 0.2.1 
language-apache 1.7.0 
language-applescript 0.2.2 (dev)
language-babel 2.54.1 
language-cson undefined (dev)
language-csv 1.1.2 
language-docker 1.1.7 
language-dots 0.0.4 
language-env 0.2.0 
language-extendscript 0.0.0 (dev)
language-generic-config 1.3.0 
language-gitignore 0.3.0 
language-ignore 1.5.2 
language-ini 1.17.0 
language-javascript-jsx 0.3.7 
language-javascript-jxa 0.2.8 (dev)
language-markdown 0.19.1 
language-marko 2.9.0 
language-postcss 1.3.0 
language-posthtml 0.0.0 (dev)
language-pug 0.0.19 
language-slm 0.5.0 
language-svg 0.9.2 
language-swift 0.5.0 
language-vue 0.21.2 
lines 0.13.1 
linter 2.2.0 
linter-alex 4.0.0 
linter-ansible-linting 1.2.2 
linter-ansible-syntax 1.1.1 
linter-coffee-variables 0.7.0 
linter-coffeelint 1.2.1 
linter-csslint 1.3.4 
linter-docker 0.1.4 
linter-doiuse 0.2.3 
linter-eslint 8.1.0 
linter-flow 5.5.0 
linter-htmlcs 0.4.0 
linter-htmlhint 1.3.1 
linter-js-yaml 1.2.6 
linter-jsonlint 1.3.0 
linter-less 2.6.0 
linter-markdown 3.1.0 
linter-package-json-validator 2.0.0 
linter-perl 0.8.0 
linter-proselint 3.2.1 
linter-pug 1.3.0 
linter-remark 2.1.0 
linter-retext 0.0.3 
linter-rubocop 0.5.3 
linter-sass-lint 1.7.4 
linter-scss-lint 3.0.4 
linter-shellcheck 1.4.3 
linter-slim 1.0.0 
linter-stylelint 3.8.0 
linter-stylint 2.2.6 
linter-swiftc 2.0.0 
linter-swiftlint 1.2.1 
linter-tidy 2.3.0 
linter-ui-default 1.6.3 
linter-write-good 0.8.0 
linter-xmllint 1.4.0 
linter-xo 0.21.2 
livereload 0.4.4 
make-executable 0.1.1 
markdown-folder 0.5.0 
markdown-pdf 1.5.0 
markdown-preview-enhanced 0.9.7 
markdown-preview-plus 2.4.10 
markdown-table-editor 0.6.0 
markdown-table-formatter 2.8.4 
markdown-toc 0.4.2 
markdown-writer 2.6.5 
medium-upload 0.1.2 
merge-conflicts 1.4.4 
mnml-syntax 0.0.0 (dev)
modular-keymaps 0.2.0 (dev)
modular-snippets 0.1.3 (dev)
monocolor-amber-syntax 1.0.0 
monocolor-lime-syntax 1.0.0 
monocolor-orange-syntax 1.0.0 
move-status-items 0.0.11 
mute-syntax undefined (dev)
narrow 0.40.0 
native-ui 0.20.4 
navigate 0.1.17 
nebula-syntax 0.4.4 
nebula-ui 0.6.0 
new-item 0.0.0 (dev)
npm-helper 0.8.9 
nucleus-dark-ui 0.10.22 
open-project-in-tower 1.4.0 
open-recent 5.0.0 (dev)
open-terminal-here 2.3.0 
package-settings 1.1.0 
package-switch 0.5.0 
pain-split 1.4.0 
pandoc 0.2.2 
pane-split-moves-tab 0.1.0 
path-hyperclick 0.3.0 
pattern-preview undefined (dev)
pdf-view undefined 
perfectionist 1.4.0 
pigments 0.39.1 
pinned-tabs 0.4.2 
platformio-ide-terminal 2.4.0 
plist-converter 0.1.6 (dev)
postcss-sorting 2.4.0 (dev)
predawn-syntax 1.0.4 (dev)
predawn-ui 1.0.2 (dev)
prettier-atom 0.28.0 
preview 0.17.0 
project-config 0.0.0 (dev)
project-manager 3.3.3 
project-plus 0.9.0 
project-templates undefined (dev)
project-viewer 1.2.0 
react-native-snippets 0.3.0 
react-redux-atom-snippets 0.7.2 
react-snippets 0.6.2 
regex-railroad-diagram 0.16.0 
release-notes 0.53.0 
rest-client 1.3.1 
ruby-bundler 0.3.0 
rulerz 0.4.3 
run-in-atom 1.0.5 
sandbox undefined (dev)
script 3.14.1 (dev)
scroll-sync 0.1.5 
sequential-command 0.3.0 
seti-icons 1.3.4 
seti-syntax 1.1.2 
seti-ui 1.6.1 
smart-quotes-plus 2.1.0 
solid-completions 2.0.0 
sort-lines 0.14.0 
transpile-preview 0.1.0 
source-preview-coffeescript undefined (dev)
source-preview-decaf 0.1.0 (dev)
source-preview-less 0.1.6 (dev)
source-preview-markdown 0.1.1 (dev)
source-preview-marko 0.1.2 (dev)
source-preview-postcss 0.1.4 (dev)
source-preview-posthtml 0.0.0 (dev)
source-preview-pug 0.2.0 (dev)
source-preview-sass 0.1.6 (dev)
source-preview-slm 0.1.0 (dev)
source-preview-stylus 0.1.7 (dev)
split-diff 1.1.2 
stylefmt 4.0.0 
Stylus 3.1.0 
Sublime-Style-Column-Selection 1.7.3 
svg-preview 0.11.0 
svgo 2.6.0 
sync-settings 0.6.0 (dev)
tab-control 0.6.9 
table-editor 0.2.2 
tablr 1.8.1 
tabs-to-spaces 1.0.3 
template-generator 0.15.0 
template-package undefined (dev)
theme-syntax  
theme-ui 0.0.0 
terminal-panel 1.14.1 (dev)
terminal-plus 0.14.5 
test-status 1.0.0 
text-manipulation 0.6.0 (dev)
things 0.0.0 
todo-show 1.9.0 
toggle-packages 1.1.0 
toggle-quotes 1.0.0 
toggler 0.3.0 
tool-bar 1.0.1 
tool-bar-git 0.0.0 (dev)
tool-bar-markdown-writer 0.3.0 
transform 0.4.0 
transpile 0.1.4 (dev)
transpile-cson 0.1.0 (dev)
css-converter 0.0.0 
transpile-decaf 0.1.3 (dev)
transpile-html2pug undefined (dev)
transpile-js2coffee undefined (dev)
transpile-lebab 0.0.0 (dev)
travis-ci-status 1.3.0 
tree-view-autoresize 2.1.0 
tree-view-breadcrumb 0.8.1 
tree-view-git-projects 0.5.0 
tree-view-git-status 1.3.0 
tree-view-search-bar 0.1.8 
tree-view-sort 0.6.0 
tree-view-symlink undefined (dev)
twilight-syntax-theme 0.1.1 
unity-ui 2.1.8 
url-utils 0.2.1 (dev)
vue-snippets 0.6.0 
wakatime 6.0.13 
warn-before-quitting 0.1.2 
wordcount undefined 
Zen 0.16.4 
zip-folder 3.2.0 
```

**Comment by cmtonkinson** [comment]
Thanks for using rulerz, and caring enough to submit a bug report! We're no longer actively using Atom day-to-day, and so haven't devoted a great deal of time to the package. If you're interested in picking up maintainership, please let me know.

**Comment by ryenski** [comment]
Out of curiosity, are you using a different editor day-to-day now? If so, which one? (Asking for a friend)

**Comment by cmtonkinson** [comment]
@mysmallidea Vim, with a small handful of plugins (through Vundle). One notable exception is a large .NET project I recently took over, for which I like MSVS 2015.

**Comment by nimamehanian** [comment]
Is there a comparable tool to Rulerz? I LOVE this utility, and am so bothered that it's not working anymore...

**Comment by n3dst4** [comment]
@nimamehanian I've started using [highlight-column](https://atom.io/packages/highlight-column). You can configure it to just show a line like rulerz, not a whole column.

**Comment by chmac** [comment]
Hopefully this issue has been fixed in the latest release, `0.5.0`. But if the highlight-column package is well maintained, it may well be a better choice.
